contracts/utils/Create2.sol

```diff
@@ -41,12 +41,29 @@ library Create2 {
         if (bytecode.length == 0) {
             revert Create2EmptyBytecode();
         }
+
+        uint256 returndataSize;
+        bytes memory returndata;
+
         /// @solidity memory-safe-assembly
         assembly {
             addr := create2(amount, add(bytecode, 0x20), mload(bytecode), salt)
+            if iszero(addr) {
+                returndataSize := returndatasize()
+                returndata := mload(0x40)
+                mstore(0x40, add(returndata, add(returndataSize, 0x20)))
```

Comment:
returndata is going to be used for the revert. It has no long term purpose, so we can do without the memory ptr update.